src/scmdata/netcdf.py

```diff
@@ -54,254 +30,250 @@
 
 
 def _var_to_nc(var):
-    return var.replace("|", "__").replace(" ", "_")
+    return var.replace("|", "_pipe_").replace(" ", "_space_")
 
 
 def _nc_to_var(var):
-    return var.replace("__", "|").replace("_", " ")
+    return var.replace("_pipe_", "|").replace("_space_", " ")
 
 
-def _get_idx(vals, v):
-    if v not in vals:
-        raise AssertionError("{} is not in {}".format(v, vals))
-
-    return np.where(vals == v)[0][0]
+def _write_nc(fname, run, dimensions, extras):
+    """
+    Low level function to write the dimensions, variables and metadata to disk
+    """
+    xr_ds = _get_xr_dataset(run, dimensions, extras)
 
+    xr_ds.attrs["created_at"] = datetime.utcnow().isoformat()
+    xr_ds.attrs["_scmdata_version"] = __version__
 
-def _get_nc_type(np_type):
-    if np_type == int:
-        return {
-            "datatype": "i8",
-        }
-    elif np_type == float:
-        return {"datatype": DEFAULT_FLOAT, "fill_value": np.nan}
+    if run.metadata:
+        xr_ds.attrs.update(run.metadata)
 
-    return {"datatype": str, "fill_value": None}
+    xr_ds.to_netcdf(fname)
```

Comment:
Can we pass through kwargs to this. Specifically, the encoding argument which controls chunking and compression

Reply:
Done (although it took a bit of thinking)